### PR TITLE
fix: Apply hardening mechanisms to controller endpoints

### LINKED ISSUE #317: Apply hardening mechanisms
We can apply the following:

- [ ] Secure Coding Practices:

Validate and sanitize user inputs to prevent XSS attacks.

- [x] Dependency Scanning:

Regularly scan and update project's dependencies to ensure that we are not using versions with known security vulnerabilities. 

- [ ] Security Headers:

Implement security headers in Spring application. For example, Content Security Policy (CSP) headers can help prevent XSS attacks by defining from which sources content can be loaded.

- [X] Static Code Analysis:

Integrate static code analysis tools into build process to identify and fix potential security issues in our code.

### PR DESCRIPTION
**Pull request by o-kopysov**
# Description

To improve the security of LPVS, we've applied hardening mechanisms according to task #317 

Fixes #317

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Code cleanup/refactoring
- [ ] Documentation update
- [ ] This change requires a documentation update
- [ ] CI system update
- [X] Test Coverage update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by userai-dev** [comment]
Tested -in container- all worked including FE-

**LPVS log:**
pvs-lpvs-1     | 2023-11-20 13:09:10.886  INFO 1 --- [nio-7896-exec-8] com.lpvs.service.LPVSStatisticsService   : userai-dev
lpvs-lpvs-1     | 2023-11-20 13:10:34.814  INFO 1 --- [           main] c.l.service.LPVSQueueProcessorService    : PROCESS Webhook id = 2
lpvs-lpvs-1     | 2023-11-20 13:10:34.822  INFO 1 --- [  LPVS-ASYNC::1] com.lpvs.service.LPVSQueueService        : GitHub Webhook processing...
lpvs-lpvs-1     | 2023-11-20 13:10:34.822  INFO 1 --- [  LPVS-ASYNC::1] com.lpvs.service.LPVSQueueService        : com.lpvs.entity.LPVSQueue@9c6c4db7
lpvs-lpvs-1     | 2023-11-20 13:36:58.135  INFO 1 --- [           main] c.l.service.LPVSQueueProcessorService    : PROCESS Webhook id = 3
lpvs-lpvs-1     | 2023-11-20 13:36:58.140  INFO 1 --- [  LPVS-ASYNC::2] com.lpvs.service.LPVSQueueService        : GitHub Webhook processing...
lpvs-lpvs-1     | 2023-11-20 13:36:58.145  INFO 1 --- [  LPVS-ASYNC::2] com.lpvs.service.LPVSQueueService        : com.lpvs.entity.LPVSQueue@fcd4663e

**PR logs:**
[License Pre-Validation Service] No license issue detected

Detected licenses:

File: stream.go
License(s): [Apache-2.0](https://www.osadl.org/fileadmin/checklists/unreflicenses/Apache-2.0.txt) (permitted)
Component: go-libvirt (internal/event/stream.go)
Matched Lines: [all](https://github.com/userai-dev/dockerapp/blob/cfb675bff4f5aaa149bc65c88488698580306bf3/stream.go)
Snippet Match: 100%


[License Pre-Validation Service] Potential license problem(s) detected

Detected licenses:

File: HashList.C
License(s): [GPL-3.0-only](https://www.osadl.org/fileadmin/checklists/unreflicenses/GPL-3.0-only.txt) (prohibited), [GPL-3.0-or-later](https://www.osadl.org/fileadmin/checklists/unreflicenses/GPL-3.0-or-later.txt) (unreviewed), [GPL-1.0-or-later](https://www.osadl.org/fileadmin/checklists/unreflicenses/GPL-1.0-or-later.txt) (unreviewed)
Component: OpenFOAM-dev (OpenFOAM/containers/HashTables/HashList/HashList.C)
Matched Lines: [all](https://github.com/userai-dev/dockerapp/blob/0324ca52f832001da90edf294535ee0b8b3cb4a8/HashList.C)
Snippet Match: 100%

**FE screenshots:**
[LPVS.pdf](https://github.com/Samsung/LPVS/files/13415539/LPVS.pdf)


[LPVS-2.pdf](https://github.com/Samsung/LPVS/files/13415541/LPVS-2.pdf)

**Review by o-konoval — APPROVED** [review]
Good job! Regarding @userai-dev comment with test results( it is my test account)-quick test was ok incl.FE. Approved.

**Review by tdrozdovsky — APPROVED** [review]

File: src/main/java/com/lpvs/controller/LPVSWebController.java [hunk 1]
```diff
@@ -20,24 +20,24 @@
...
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.HtmlUtils;
 
 import java.sql.Timestamp;
 import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (m-rudyk) [inline src/main/java/com/lpvs/controller/LPVSWebController.java:40]: I suggest, we keep explicitly listing all imports.

**Review by m-rudyk — APPROVED** [review]
One comment, please check

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**
- Coverage: 93.295%
- Branches: 80.943%

**Pull request merged**